utils/ada_helpers.py

```diff
@@ -0,0 +1,451 @@
+            fee_amount = DEFAULT_NATIVE_TX_FEE
+
+        for utxo in all_utxos:
+            if len(utxo.tokens) > 0:
+                continue
+            collected_utxos.append(utxo)
+            collected_native_amount += utxo.native_amount
+            if collected_native_amount >= net_amount + fee_amount:
+                change_amount = collected_native_amount - net_amount - fee_amount
+                if (change_amount == 0) or (change_amount >= MIN_UTXO_VALUE_ADA_ONLY):
+                    break
+
+        if collected_native_amount < net_amount + fee_amount:
+            raise Exception(f'Insufficient Balance in account ({collected_native_amount} < {net_amount + fee_amount})')
+
+        change_amount = collected_native_amount - net_amount - fee_amount
+        if 0 < change_amount < MIN_UTXO_VALUE_ADA_ONLY:
+            raise Exception(f'Change output below minimum {change_amount} < {MIN_UTXO_VALUE_ADA_ONLY}')
+
+        change_address = self.get_base_address(0)
+        serialized_tx_payload, deserialized_tx_payload = self.__build_tx_payload(
+            to_address, net_amount, collected_utxos, change_amount, change_address.address,
+            fee_amount, self.__get_ttl()
+        )
+
+        signing_payload = CardanoWallet.__get_signing_payload(serialized_tx_payload)
+        sigs = self.__sign_tx_payload(fpriv, collected_utxos, signing_payload)
+        signed_tx, _ = CardanoWallet.__embed_sigs_in_tx(deserialized_tx_payload, sigs)
+        return self.__gql_client.submit_tx(signed_tx)
+
+
+    def __get_utxos_for_account(self) -> List[CardanoUTxO]:
+        return self.__gql_client.get_utxos(
+            self.get_account_addresses(), True
+        )
+
+
+    def __find_max_address_index(self, address_pool_gap: int) -> int:
+        if address_pool_gap <= 0:
+            raise Exception(f'Invalid address_pool_gap value of {address_pool_gap}')
+
+        curr_gap = 0
+        curr_address_idx = self.__min_address_index
+        while curr_gap < address_pool_gap:
+            curr_balance = self.__get_balance_for_addresses([
+                self.get_base_address(curr_address_idx),
+                self.get_enterprise_address(curr_address_idx)
+            ])
+
+            curr_gap = curr_gap + 1 if curr_balance.is_empty() else 0
+            curr_address_idx += 1
+
+        max_index = curr_address_idx - address_pool_gap - 1
+        if max_index < 0:
+            raise Exception(f'Failed to find max index for address_pool_gap of {address_pool_gap} '
+                            f'(is the wallet empty?)')
+
+        return max_index
+
+
+    def __get_balance_for_addresses(self, addresses: List[CardanoAddress]) -> CardanoBalanceMap:
+        balances_map = CardanoBalanceMap()
+        for utxo in self.__gql_client.get_utxos(addresses, True):
+            balances_map.increase_balance('ADA', utxo.native_amount)
+            for token in utxo.tokens:
+                balances_map.increase_balance(token.name, token.amount)
```

Comment:
I think we shouldn't add it to the overall balance. What is the output in graphql if you query paymentAddresses? I think it's separate per asset.
However, as we don't support tokens yet, we shouldn't receive any utxo.tokens,
consider throwing unsupported for now.

Reply:
Not sure if I agree, but lets leave it for the next iteration. 
I don't think we should throw, we can allow this information to be retrieved.

Reply:
Even when we do support tokes, it should be separated by asset, as it's not an ADA balance